wandb/lib/lib.py

```diff
@@ -0,0 +1,70 @@
+import ctypes
+import pathlib
+import platform
+from typing import Optional
+
+
+class Lib:
+    def __init__(self) -> None:
+        self._lib: Optional[ctypes.CDLL] = None
+        self.init_sentry()
+
+    @property
+    def lib(self) -> ctypes.CDLL:
+        if self._lib is not None:
+            return self._lib
+        dylib_ext = "dll" if platform.system().lower() == "windows" else "so"
+        path_lib = (
+            pathlib.Path(__file__).parent.parent / "bin" / f"libwandb.{dylib_ext}"
+        )
+        self._lib = ctypes.CDLL(str(path_lib))
+        return self._lib
+
+    def setup(self, core_path: str) -> str:
+        self.lib.Setup.argtypes = [ctypes.c_char_p]
+        self.lib.Setup.restype = ctypes.c_char_p
```

Comment:
is it correct to do this on every call? shouldn't we only do this when loading the library?

Reply:
good point, fixed